### ISSUE TITLE
Invalid type after assigning an array to the object

### ISSUE THREAD

**Issue by MateuszKlatecki** [nanoframework/Home#484]
## Details about Problem

**nanoFramework area:** ( mscorelib or compiler)

## Description

After assign array (no matter what type) to object its type cannot be determined.

Sample code:
```cs
using System;

namespace nfTest
{
    public class Program
    { 
        class MyType
        {

        }
        static void Main(string[] args)
        {
            var o = new int[] { 1234 };

            if (o is byte)
                Console.WriteLine("Byte");
            else
                Console.WriteLine("Not Byte");

            if (o is Array)
                Console.WriteLine("Array");
            else
                Console.WriteLine("Not Array");

            object obj = o;

            Console.WriteLine("--=== After assign ===--");

            if (obj is byte)
                Console.WriteLine("Byte");

            if (obj is Array)
                Console.WriteLine("Array");

            if (obj is Program.MyType)
                Console.WriteLine("MyType");

        }
    }
}

```
Console output
```console
Not Byte
Array
--=== After assign ===--
Byte
Array
MyType
```

Interestingly, if it is not an array, the type after the assignment is recognized correctly

Of course, in normal .net on pc everything works:

```console
Not Byte
Array
--=== After assign ===--
Array
```

## Expected behaviour

After assigning the table to the object, it is still possible to recognize the type

**Comment by josesimoes** [comment]
@MateuszKlatecki when you have the opportunity can you please run this test again? It might have been fixed with nanoframework/nf-interpreter#1547...

**Comment by MateuszKlatecki** [comment]
Unfortunately, this change does not fix my case.
